compose/service.py

```diff
@@ -678,10 +678,21 @@ def labels(self, one_off=False):
         ]
 
     def can_be_scaled(self):
-        for port in self.options.get('ports', []):
-            if ':' in str(port):
+        # TODO: this function should ideally move to docker.utils.ports
+        # get rid of it once its accepted into docker-py
+        def has_external_port(external_binding):
+            if external_binding is None:
```

Comment:
If we do what I have proposed in https://github.com/docker/compose/pull/1659/files#r34191487 then we don't have to put this `has_external_bindings()` function in docker-py as a public function at all, we can simply inspect what docker-py's `split_port()` provides us.